src/utils/ar_usd_oracle.ts

```diff
@@ -0,0 +1,65 @@
+import type { FiatOracle } from './fiat_oracle';
+// import CoinGecko from 'coingecko-api';
+import type { CoinGeckoPriceRequestParams, CoinGeckoPriceResponseData, CoinGeckoResponse } from './coingecko_types';
+
+const AR_COIN_ID = 'arweave';
+const USD_COIN_ID = 'usd';
+const pollingIntervalTime = 1000 * 60 * 15; // 15min expressed in in ms
+
+export class AR_USDOracle implements FiatOracle {
+	private cachedPrice = 0;
+	private cacheTimestamp = 0;
+	private syncPromise?: Promise<void>;
+
+	// constructor(private readonly client = new CoinGecko()) {}
+
+	private get currentlyFetchingPrice() {
+		return !!this.syncPromise;
+	}
+
+	public async ping(): Promise<boolean> {
+		// const response = await this.client.ping();
+		const response = await fetch('https://api.coingecko.com/api/v3/ping');
+		return response.ok;
+	}
+
+	public async getUSDPriceForARAmount(ARAmount: number): Promise<number> {
```

Comment:
I've already commented on how I think this API should look in the interface declaration, but just another note: the parameter name should start with lowercase, i.e. `arAmount`